ctapipe/tools/apply_models.py

```diff
@@ -0,0 +1,234 @@
+            mono_predictions = self._apply(self.energy_regressor, "energy")
+            self.log.info(
+                "Combining telescope-wise energy predictions to array event prediction"
+            )
+            self._combine(self.energy_regressor.stereo_combiner, mono_predictions)
+
+        if self.apply_classifier:
+            self.log.info("Applying classifier to telescope events")
+            mono_predictions = self._apply(self.classifier, "classification")
+            self.log.info(
+                "Combining telescope-wise particle id predictions to array event prediction"
+            )
+            self._combine(self.classifier.stereo_combiner, mono_predictions)
+
+    def _apply(self, reconstructor, parameter):
+        prefix = reconstructor.model_cls
+
+        tel_tables = []
+
+        desc = f"Applying {reconstructor.__class__.__name__}"
+        unit = "telescope"
+        for tel_id, tel in tqdm(self.loader.subarray.tel.items(), desc=desc, unit=unit):
+            if tel not in reconstructor._models:
+                self.log.warning(
+                    "No model in %s for telescope type %s, skipping tel %d",
+                    reconstructor,
+                    tel,
+                    tel_id,
+                )
+                continue
+
+            table = self.loader.read_telescope_events([tel_id])
+            if len(table) == 0:
+                self.log.warning("No events for telescope %d", tel_id)
+                continue
+
+            table.remove_columns([c for c in table.colnames if c.startswith(prefix)])
+
+            predictions = reconstructor.predict_table(tel, table)
+            table = hstack(
+                [table, predictions],
+                join_type="exact",
+                metadata_conflicts="ignore",
+            )
+            write_table(
+                table[["obs_id", "event_id", "tel_id"] + predictions.colnames],
+                self.output_path,
+                f"/dl2/event/telescope/{parameter}/{prefix}/tel_{tel_id:03d}",
+                mode="a",
+                overwrite=self.overwrite,
+            )
+            tel_tables.append(table)
+
+        if len(tel_tables) == 0:
+            raise ValueError("No predictions made for any telescope")
+
+        return vstack(tel_tables)
+
+    def _combine(self, combiner, mono_predictions):
+        stereo_predictions = combiner.predict_table(mono_predictions)
+
+        trafo = TelListToMaskTransform(self.loader.subarray)
+        for c in filter(
+            lambda c: c.name.endswith("telescopes"),
+            stereo_predictions.columns.values(),
+        ):
+            stereo_predictions[c.name] = np.array([trafo(r) for r in c])
+            stereo_predictions[c.name].description = c.description
+
+        write_table(
+            stereo_predictions,
+            self.output_path,
+            f"/dl2/event/subarray/{combiner.property}/{combiner.prefix}",
```

Comment:
Since `combiner.property` is used in the table path here, it would be helpful to rename the property "direction" to "geometry" so the disp geometry prediction gets saved at the correct place.

Reply:
But I can also change that in the disp PR

Reply:
done